torch/csrc/utils/tensor_new.cpp

```diff
@@ -321,14 +322,14 @@ static Tensor set_requires_grad(Tensor self, bool requires_grad) {
 
 Tensor new_tensor(const Type& type, PyObject* args, PyObject* kwargs) {
   static PythonArgParser parser({
-    "new_tensor(Tensor other, *, Type dtype=None, bool requires_grad=False)",
+    "new_tensor(Tensor other, *, Type dtype=None, int64_t? device=-1, bool requires_grad=False)",
```

Comment:
It would be nice to combine these signatures to`PyObject *data, ...` and do the tensor check internally. (Mostly have the same kwarg, so that we can document a single overload)